SPECS/golang/CVE-2025-47906-1.18.patch

```diff
@@ -0,0 +1,137 @@
+
+diff --git a/src/os/exec/exec.go b/src/os/exec/exec.go
+index 056a7e0..6711197 100644
+--- a/src/os/exec/exec.go
++++ b/src/os/exec/exec.go
+@@ -799,3 +799,22 @@ func addCriticalEnv(env []string) []string {
+ 	}
+ 	return append(env, "SYSTEMROOT="+os.Getenv("SYSTEMROOT"))
+ }
++
++// ErrDot indicates that a path lookup resolved to an executable
++// in the current directory due to ‘.’ being in the path, either
++// implicitly or explicitly. See the package documentation for details.
++//
++// Note that functions in this package do not return ErrDot directly.
++// Code should use errors.Is(err, ErrDot), not err == ErrDot,
++// to test whether a returned error err is due to this condition.
++var ErrDot = errors.New("cannot run executable found relative to current directory")
++
++// validateLookPath excludes paths that can't be valid
++// executable names. See issue #74466 and CVE-2025-47906.
++func validateLookPath(s string) error {
++	switch s {
++	case "", ".", "..":
++		return ErrNotFound
++	}
++	return nil
++}
+diff --git a/src/os/exec/exec_test.go b/src/os/exec/exec_test.go
+index 0be8c6c..b459915 100644
+--- a/src/os/exec/exec_test.go
++++ b/src/os/exec/exec_test.go
+@@ -1098,3 +1098,16 @@ func TestNoPath(t *testing.T) {
+ 		t.Errorf("new(Cmd).Start() = %v, want %q", err, want)
+ 	}
+ }
++
++func TestLookPathInvalidNames(t *testing.T) {
++    invalids := []string{"", ".", ".."}
++    for _, name := range invalids {
++        path, err := exec.LookPath(name)
++        if err == nil {
++            t.Errorf("LookPath(%q): expected error, got nil", name)
++        }
++        if path != "" {
++            t.Errorf("LookPath(%q): expected empty path, got %q", name, path)
++        }
++    }
++}
+diff --git a/src/os/exec/lp_plan9.go b/src/os/exec/lp_plan9.go
+index e8826a5..a4fb95d 100644
+--- a/src/os/exec/lp_plan9.go
++++ b/src/os/exec/lp_plan9.go
+@@ -32,6 +32,10 @@ func findExecutable(file string) error {
+ // directly and the path is not consulted.
+ // The result may be an absolute path or a path relative to the current directory.
+ func LookPath(file string) (string, error) {
++	if err := validateLookPath(file); err != nil {
++		return "", &Error{file, err}
++	}
++
+ 	// skip the path lookup for these prefixes
+ 	skip := []string{"/", "#", "./", "../"}
+ 
+diff --git a/src/os/exec/lp_unix.go b/src/os/exec/lp_unix.go
+index 38b9fc7..9d6bf54 100644
+--- a/src/os/exec/lp_unix.go
++++ b/src/os/exec/lp_unix.go
+@@ -37,6 +37,11 @@ func LookPath(file string) (string, error) {
+ 	// (only bypass the path if file begins with / or ./ or ../)
+ 	// but that would not match all the Unix shells.
+ 
++	if err := validateLookPath(file); err != nil {
++		return "", &Error{file, err}
++	}
++
++
+ 	if strings.Contains(file, "/") {
+ 		err := findExecutable(file)
+ 		if err == nil {
+diff --git a/src/os/exec/lp_windows.go b/src/os/exec/lp_windows.go
+index e7a2cdf..7a1d6fb 100644
+--- a/src/os/exec/lp_windows.go
++++ b/src/os/exec/lp_windows.go
```

Comment:
Note: Upstream has another instance of using the `validateLookPath` method. But since it's in the windows part, ignoring it.